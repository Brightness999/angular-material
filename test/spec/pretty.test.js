/**
 * @fileOverview Test normal logging.
 */
const Logality = require('../..');
const { sinkPretty, stubLogality } = require('../lib/tester.lib');

describe('Pretty Logging', () => {
  stubLogality();

  test('Will pretty log expected JSON properties', (done) => {
    const logality = new Logality({
      prettyPrint: true,
      appName: 'testLogality',
      wstream: sinkPretty((chunk) => {
        expect(chunk).toMatchSnapshot();
        done();
      }),
    });

    const log = logality.get();

    log('info', 'hello world');
  });


  test('Will pretty log an object in context', (done) => {
    const logality = new Logality({
      prettyPrint: true,
      appName: 'testLogality',
      wstream: sinkPretty((chunk) => {
        expect(chunk).toMatchSnapshot();
        done();
      }),
    });

    const log = logality.get();

    log('info', 'hello world', { custom: { a: 1, b: 2 } });
  });
});
