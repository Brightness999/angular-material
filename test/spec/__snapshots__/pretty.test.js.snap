// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pretty Logging Will pretty log an object in context 1`] = `
"[37m[2018-05-18T16:25:57.815Z][39m [34mâ„¹ info[39m [4m[32m/test/spec/pretty.test.js[39m[24m - [34mhello world[39m
{
  \\"context\\": {
    \\"custom\\": {
      \\"a\\": 1,
      \\"b\\": 2
    }
  }
}
"
`;

exports[`Pretty Logging Will pretty log expected JSON properties 1`] = `
"[37m[2018-05-18T16:25:57.815Z][39m [34mâ„¹ info[39m [4m[32m/test/spec/pretty.test.js[39m[24m - [34mhello world[39m
"
`;
